CourseManager/Courses/Course 4 UI and Animations in SwiftUI/Views/Main/HomeViewWeather.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  HomeViewWeather.swift
+//  CourseManager
+//
+//  Created by Summer Crow on 2022-11-09.
+//
+
+import SwiftUI
+import BottomSheet
+
+enum BottomSheetPosition: CGFloat, CaseIterable {
+    case top = 0.83
+    case middle = 0.385
+}
+
+struct HomeViewWeather: View {
+    @State var bottomSheetPosition: BottomSheetPosition = .middle
+    @State var bottomSheetTranslation: CGFloat = BottomSheetPosition.middle.rawValue
+    @State var hasDragged: Bool = false
+
+    var bottomSheetTranslationProrated: CGFloat {
+        (bottomSheetTranslation - BottomSheetPosition.middle.rawValue) / (BottomSheetPosition.top.rawValue - BottomSheetPosition.middle.rawValue)
+    }
+
+
+    var body: some View {
+        NavigationView {
+            GeometryReader { geometry in
+                let screenHeight = geometry.size.height + geometry.safeAreaInsets.top + geometry.safeAreaInsets.bottom
+                let imageOffset = screenHeight + 36
+
+                ZStack {
+                    Color.weatherBackground
+                        .ignoresSafeArea()
+                    Image("Background")
+                        .resizable()
+                        .ignoresSafeArea()
+                        .offset(y: -bottomSheetTranslationProrated * imageOffset)
+
+                    Image("House")
+                        .frame(maxHeight: .infinity, alignment: .top)
+                        .padding(.top, 257)
+                        .offset(y: -bottomSheetTranslationProrated * imageOffset)
+
+                    VStack(spacing: -10 * (1 - bottomSheetTranslationProrated)) {
+                        Text("Montreal")
+                            .font(.largeTitle)
+
+                        VStack {
+                            Text(attributedString)
+                            Text("H:24°   L:18°")
+                                .font(.title3.weight(.semibold))
+                                .opacity(1 - bottomSheetTranslationProrated)
+
+                        }
+
+                        Spacer()
+                    }
+                    .padding(.top, 51)
+                    .offset(y: -bottomSheetTranslationProrated * 46)
+
+                    BottomSheetView(position: $bottomSheetPosition) {
+                    } content: {
+                        ForecastView(bottomSheetTranslationProrated: bottomSheetTranslationProrated)
+                    }
+                    .onBottomSheetDrag { translation in
+                        bottomSheetTranslation = translation / screenHeight
+                        withAnimation(.easeInOut) {
+                               if bottomSheetPosition == BottomSheetPosition.top {
```

Comment:
hasDragged = bottomSheetPosition == .top?

Reply:
I've updated this to: 
bottomSheetPosition == .top ? (hasDragged = true) : (hasDragged = false)

Reply:
It isn't the best though

Reply:
Oh sorry, you mean this: 
hasDragged = bottomSheetPosition == .top ? true : false